pennylane/ops/op_math/controlled_ops.py

```diff
@@ -130,6 +123,9 @@ class ControlledQubitUnitary(ControlledOp):
     grad_method = None
     """Gradient computation method."""
 
+    def _flatten(self):
+        return (self.base,), (self.wires, tuple(self.control_values), self.work_wires)
+
     @classmethod
     def _unflatten(cls, data, metadata):
         return cls(data[0], metadata[0], control_values=metadata[1], work_wires=metadata[2])
```

Comment:
But isn't `self.base` a `QubitUnitary` operator, and passing a `QubitUnitary` is deprecated? 
```suggestion
    def _flatten(self):
        return (self.base.data[0],), (self.wires, tuple(self.control_values), self.work_wires)

    @classmethod
    def _unflatten(cls, data, metadata):
        return cls(data[0], metadata[0], control_values=metadata[1], work_wires=metadata[2])
```

Reply:
For flatten and unflatten, we mostly care about being able to do:
```
copied_op = op._unflatten(*op._flatten())
``

Reply:
```
>>> op = qml.ControlledQubitUnitary(np.eye(2), (1,2,3))
>>> qml.ops.functions.assert_valid(op, skip_differenitation=True)
[pennylane/ops/op_math/controlled_ops.py:225](/ops/op_math/controlled_ops.py#line=224): PennyLaneDeprecationWarning: QubitUnitary input to ControlledQubitUnitary is deprecated and will be removed in v0.42. Instead, please use a full matrix as input.
  warnings.warn(
```
I wonder how that was being missed by the test suite 🤔

Reply:
> ```
> >>> op = qml.ControlledQubitUnitary(np.eye(2), (1,2,3))
> >>> qml.ops.functions.assert_valid(op, skip_differenitation=True)
> [pennylane/ops/op_math/controlled_ops.py:225](/ops/op_math/controlled_ops.py#line=224): PennyLaneDeprecationWarning: QubitUnitary input to ControlledQubitUnitary is deprecated and will be removed in v0.42. Instead, please use a full matrix as input.
>   warnings.warn(
> ```
> 
> I wonder how that was being missed by the test suite 🤔

I'm also confused. Somehow previous tests never complain until I tried to deprecate; then have to overlaod the flatten